src/service_daemon.rs

```diff
@@ -770,6 +770,18 @@ struct IntfSock {
     sock: Socket,
 }
 
+impl IntfSock {
+    /// Returns the interface index and the associated ip version
+    fn intf_idx_ip_ver(&self) -> (u32, u8) {
```

Comment:
IMO, define a new `struct` for the returned tuple would be good, i.e. improve the readability and rely less on the comments. It will also help in the later parts of the changes.

Reply:
Yes, I agree. I'll make it a struct instead. Perhaps with a method is_valid() that checks just returns sellf.index > 0;